openmdao/drivers/pyoptsparse_driver.py

```diff
@@ -255,11 +260,25 @@ def objfunc(self, dv_dict):
 
             # Get the objective function evaluations
             for name, obj in iteritems(self.get_objectives()):
-                func_dict[name] = obj
+                if nproc > 1:
+                    owner = system._owning_ranks[name]
```

Comment:
The owner is the lowest rank process that has that variable as a local variable.